include/dlaf/interface/cholesky.h

```diff
@@ -0,0 +1,105 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2023, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <mpi.h>
+#include <dlaf/interface/cholesky.h>
+
+#include <pika/init.hpp>
+#include <dlaf/factorization/cholesky.h>
+#include <dlaf/interface/blacs.h>
+#include <dlaf/interface/utils.h>
+#include <dlaf/matrix/matrix.h>
+#include <dlaf/matrix/matrix_mirror.h>
+
+namespace dlaf::interface {
+
+template <typename T>
+using MatrixMirror = dlaf::matrix::MatrixMirror<T, dlaf::Device::Default, dlaf::Device::CPU>;
+
+template <typename T>
+void pxpotrf(char uplo, T* a, int m, int n, int mb, int nb, int lld, const MPI_Comm& communicator,
+             int nprow, int npcol) {
+  pika::resume();
```

Comment:
I would not support calling multiple algorithm with the C interface. It anyway needs a synchronization as the matrix object should be recreated at each call.

Reply:
Right, makes sense. In any case, I would simply document that we initialize and start the pika runtime in `dlaf_initialize`, resume and suspend the runtime around algorithm calls, and stop the pika runtime in `dlaf_finalize`, since this affects composability with other libraries potentially using pika. I know it's not a concern right now (or ever!) but I'd just like it to be explicitly documented.